cluster/repository/identity_cluster.go

```diff
@@ -102,29 +103,29 @@ func (m *GormIdentityClusterRepository) Create(ctx context.Context, c *IdentityC
 	return nil
 }
 
-// Delete removes the identity/cluster relationship identified by the given `identityID` and `clusterID`
-func (m *GormIdentityClusterRepository) Delete(ctx context.Context, identityID, clusterID uuid.UUID) error {
+// Delete removes the identity/cluster relationship identified by the given `identityID` and `clusterURL`
+func (m *GormIdentityClusterRepository) Delete(ctx context.Context, identityID uuid.UUID, clusterURL string) error {
 	defer goa.MeasureSince([]string{"goa", "db", "identity_cluster", "delete"}, time.Now())
 
-	c := IdentityCluster{IdentityID: identityID, ClusterID: clusterID}
-
-	result := m.db.Delete(&c)
+	result := m.db.Exec(fmt.Sprintf(`delete from %[1]s where identity_id = ? 
+		and cluster_id = (select cluster_id from %[2]s where url = ?)`,
+		IdentityCluster{}.TableName(), Cluster{}.TableName()), identityID.String(), clusterURL)
 
 	if result.Error != nil {
 		log.Error(ctx, map[string]interface{}{
-			"cluster_id":  c.ClusterID.String(),
-			"identity_id": c.IdentityID.String(),
+			"cluster_url": clusterURL,
+			"identity_id": identityID.String(),
 			"err":         result.Error,
 		}, "unable to delete the identity cluster")
 		return errs.WithStack(result.Error)
 	}
 	if result.RowsAffected == 0 {
-		return errors.NewNotFoundErrorFromString(fmt.Sprintf("nothing to delete: identity cluster not found (identityID:\"%s\", clusterID:\"%s\")", identityID, clusterID))
+		return errors.NewNotFoundErrorFromString(fmt.Sprintf("nothing to delete: identity cluster not found (identityID:\"%s\", clusterURL:\"%s\")", identityID.String(), clusterURL))
```

Comment:
With this, we are returning a not found error, but we don't have any context what's wrong? I mean is it because of a cluster with given URL not found or identity is not linked to a cluster which exists not found. some small hint in the logging will help to understand what's going wrong

Reply:
@dipak-pawar are you suggesting that we should run a secondary query in case where no row was deleted, to check if a cluster with the given URL exists, for example?

Reply:
@xcoulon I am fine with anything until we have some logs for what's going wrong..

Reply:
ok, so let me add this, and even customize the message returned in the response, then

Reply:
@dipak-pawar addressed in 36118ca